MSYS: unrecognized option '--hash-style=gnu'

First things first: Thanks for the fix: https://github.com/BobBuildTool/bob/issues/395 👍 

now there is a problem with the basement: `unrecognized option '--hash-style=gnu'`

https://github.com/BobBuildTool/basement/blob/52ea909a6cdb2000bf4ea72728667f592da5c582/default.yaml#L43

```
CMake Error at C:/msys64/home/mhaase/ws/fdt.bobrecipes/dev/dist/devel/win/cmake/1/workspace/usr/share/cmake-3.17/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "C:/msys64/usr/bin/gcc.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/msys64/home/mhaase/ws/fdt.bobrecipes/dev/build/devel/parasoft-cpptest/x86_64-pc-msys/1/workspace/build/CMakeFiles/CMakeTmp

    Run Build Command(s):C:/msys64/home/mhaase/ws/fdt.bobrecipes/dev/dist/devel/win/ninja/1/workspace/ninja.exe cmTC_a1873 && [1/2] Building C object CMakeFiles/cmTC_a1873.dir/testCCompiler.c.obj
    [2/2] Linking C executable cmTC_a1873.exe
    FAILED: cmTC_a1873.exe
    cmd.exe /C "cd . && C:\msys64\usr\bin\gcc.exe -Os -pipe -fPIC -DCPPTEST_SHARED=0 -DCPPTEST_NO_THREADS -DCPPTEST_FILE_COMMUNICATION  -Wl,-O1 -Wl,--hash-style=gnu CMakeFiles/cmTC_a1873.dir/testCCompiler.c.obj  -o cmTC_a1873.exe -Wl,--out-implib,libcmTC_a1873.dll.a -Wl,--major-image-version,0,--minor-image-version,0   && cd ."
    /usr/lib/gcc/x86_64-pc-msys/10.2.0/../../../../x86_64-pc-msys/bin/ld: unrecognized option '--hash-style=gnu'
    /usr/lib/gcc/x86_64-pc-msys/10.2.0/../../../../x86_64-pc-msys/bin/ld: use the --help option for usage information
    collect2: error: ld returned 1 exit status
    ninja: build stopped: subcommand failed.
```

--- a/default.yaml
+++ b/default.yaml
@@ -40,7 +40,7 @@ environment:
     # position independent code in case static and dynamic libraries are mixed.
     CFLAGS:     "-Os -pipe -fPIC"
     CXXFLAGS:   "-Os -pipe -fPIC"
-    LDFLAGS:    "-Wl,-O1 -Wl,--hash-style=gnu"
+    LDFLAGS:    "-Wl,-O1"
 
 archive:
     backend: http

--- a/recipes/devel/host-compat-toolchain.yaml
+++ b/recipes/devel/host-compat-toolchain.yaml
@@ -87,3 +87,4 @@ provideVars:
     # target-toolchain. ARCH is unchaged, though.
     AUTOCONF_BUILD: "$AUTOCONF_TARGET"
     AUTOCONF_HOST:  "$AUTOCONF_TARGET"
+    LDFLAGS: "$LDFLAGS  -Wl,--hash-style=gnu"

